Strip all hop by hop headers on request and response

kmoe opened the pull request:
fixes #9

jsangilve commented:
Hi! I'm not a reviewer, but here some minor improvements I think could be done to this PR:



There, `remove_hop_by_hop_headers` already considers the `transfer-encoding` header, [this](https://github.com/tallarium/reverse_proxy_plug/pull/12/files#diff-fa99bbfa7b2f8e822a5f1402a90ee63aR85) could be reduced to:

```elixir
 %HTTPoison.AsyncHeaders{headers: headers} ->
        headers
        |> Enum.map(fn {header, value} -> {header |> String.downcase(), value} end)
        |> remove_hop_by_hop_headers
        |> Enum.reject(fn {header, _} -> header == "content-length" end)
        |> Enum.concat([{"transfer-encoding", "chunked"}])
```

Also, the function `remove_hop_by_hop_headers` has a side effect besides removing `hop_by_hop` headers: it's transforming all the headers to lowercase. I would suggest not to this here, but either do it explicitly before calling `remove_hop_by_hop_headers` (like in the code example above `Enum.map(fn {header, value} -> {header |> String.downcase(), value} end)`) or in a separate function, let's say `normalize_headers`.

In any case, I'd suggest to slightly change `remove_hop_by_hop_headers` to:
```elixir
defp remove_hop_by_hop_headers(headers) do
    hop_by_hop_headers = [
      "te",
      "transfer-encoding",
      "trailer",
      "connection",
      "keep-alive",
      "proxy-authenticate",
      "proxy-authorization",
      "upgrade"
    ]

    headers
    |> Enum.reject(fn {header, _} -> Enum.member?(hop_by_hop_headers, String.downcase(header)) end)
end
```

Great work guys! we're passing a ton of request through an app using this proxy.

kmoe commented:
Thanks for the helpful comments @jsangilve. I've removed the duplication when normalising headers and simplified the code using a single `normalize_headers` function. 
Since Plug.Conn will raise a `Plug.Conn.InvalidHeaderError` if it encounters a header containing uppercase letters during testing, I think we can downcase them first and assume in subsequent functions that they are lowercase.
